samples/contrib/aws-samples/mnist-kmeans-sagemaker/kmeans-hpo-pipeline.py

```diff
@@ -115,7 +115,7 @@ def hpo_test(region='us-west-2',
         checkpoint_config=checkpoint_config,
         tags=tags,
         role=role_arn,
-    ).apply(use_aws_secret('aws-secret', 'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY'))
```

Comment:
I am worried we are completely removing `aws-secret` example with this PR.  Is this going to work on self-hosted clusters on AWS?  I am not sure IAM Roles for Service Accounts are working with any self-hosted cluster.  Is it possible to keep `use_aws_secret` as an additional option?  Having the ability to run this example on non-EKS clusters would be hugely valuable.

Reply:
done